pkg/snapstore/gcs_snapstore.go

```diff
@@ -68,6 +71,27 @@ func NewGCSSnapStore(config *brtypes.SnapstoreConfig) (*GCSSnapStore, error) {
 			opts = append(opts, option.WithEndpoint(endpoint))
 		}
 	}
+	// chunkSuffix is used when fakeGCS is enabled, as the emulator does not support having a file and a folder with the same name
+	// So a "@chunk" suffix is added to the chunks folder to avoid conflict with the composite object
+	// Refer to this github issue for more details: https://github.com/fsouza/fake-gcs-server/issues/1434
+	var chunkSuffix string
+	if fakegcsenabled, ok := os.LookupEnv(envEmulatorEnabled); ok {
+		isEmulator, err := strconv.ParseBool(fakegcsenabled)
+		if err != nil {
+			return nil, fmt.Errorf("value %s for environment variable %s could not be parsed", fakegcsenabled, envEmulatorEnabled)
+		}
+		if isEmulator {
+			var endpoint string
+			if endpoint, ok = os.LookupEnv(envStorageAPIEndpoint); !ok {
+				return nil, fmt.Errorf("GCS Emulator host is not set")
+			}
+			endpoint = strings.TrimSpace(endpoint)
+			os.Setenv("STORAGE_EMULATOR_HOST", strings.TrimPrefix(endpoint, "http://"))
+			opts = append(opts, option.WithEndpoint(endpoint))
+			opts = append(opts, option.WithoutAuthentication())
+			chunkSuffix = "@chunk"
+		}
+	}
```

Comment:
Please pull this code of checking whether a GCS emulator is being used, into a separate function, to keep the code flow as simple as possible. Additionally,

Reply:
Done.